### PR TITLE
[Merged by Bors] - fix default network handling

### PR DESCRIPTION
**Pull request by divagant-martian**
## Issue Addressed
#1992 and #1987, and also to be considered a continuation of #1751

## Proposed Changes
many changed files but most are renaming to align the code with the semantics of `--network` 
- remove the `--network` default value (in clap) and instead set it after checking the `network` and `testnet-dir` flags
- move `eth2_testnet_config` crate to `eth2_network_config`
- move `Eth2TestnetConfig` to `Eth2NetworkConfig`
- move `DEFAULT_HARDCODED_TESTNET` to `DEFAULT_HARDCODED_NETWORK`
- `beacon_node`s `get_eth2_testnet_config` loads the `DEFAULT_HARDCODED_NETWORK` if there is no network nor testnet provided
- `boot_node`s config loads the config same as the `beacon_node`, it was using the configuration only for preconfigured networks (That code is ~1year old so I asume it was not intended)
- removed a one year old comment stating we should try to emulate `https://github.com/eth2-clients/eth2-testnets/tree/master/nimbus/testnet1` it looks outdated (?)
- remove `lighthouse`s `load_testnet_config` in favor of `get_eth2_network_config` to centralize that logic (It had differences)
- some spelling

## Additional Info
Both the command of #1992 and the scripts of #1987 seem to work fine, same as `bn` and `vc`

### CODE REVIEW

File: boot_node/src/config.rs [hunk 1]
```diff
@@ -76,7 +76,7 @@ impl<T: EthSpec> TryFrom<&ArgMatches<'_>> for BootNodeConfig<T> {
             let spec = eth2_network_config
                 .yaml_config
                 .as_ref()
-                .ok_or_else(|| "The network directory must contain a spec config".to_string())?
+                .ok_or("The testnet directory must contain a spec config")?
```
Comment (AgeManning) [inline boot_node/src/config.rs:79]: We've renamed this to `network` right? The `--network` command. I see it does conflict with the actual networking directory, but the text here is correct right?

**Review by AgeManning — COMMENTED** [review]
This mainly looks like a lot of renaming. And seems good to me

File: boot_node/src/config.rs (hunk 1 above)
Comment (divagant-martian) [reply boot_node/src/config.rs]: you are right, done

**Review by AgeManning — APPROVED** [review]

**Pull request closed without merging**